_includes/feed.html

```diff
@@ -4,23 +4,32 @@
 assign limit = include.limit | default: site.paginate | default: 10
 %}{%
 assign title = include.title | default: site.title
+%}{%
+assign url_prefix = site.url | append: site.baseurl | append: '/'
 %}<?xml version="1.0" encoding="UTF-8"?>
 <rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom">
   <channel>
     <title>{{ title | xml_escape }}</title>
     <description>{{ site.description | xml_escape }}</description>
-    <link>{{ site.url }}{{ site.baseurl }}/</link>
-    <atom:link href="{{ "/feed.xml" | prepend: site.baseurl | prepend: site.url }}" rel="self" type="application/rss+xml"/>
+    <link>{{ url_prefix }}</link>
+    <atom:link href="{{ "/feed.xml" | prepend: url_prefix }}" rel="self" type="application/rss+xml"/>
     <pubDate>{{ site.time | date_to_rfc822 }}</pubDate>
     <lastBuildDate>{{ site.time | date_to_rfc822 }}</lastBuildDate>
     <generator>Jekyll v{{ jekyll.version }}</generator>
     {% for post in posts limit: limit %}
       <item>
         <title>{{ post.title | xml_escape }}</title>
-        <description>{{ post.content | markdownify | xml_escape }}</description>
+        <description>
+          {{ post.content
+            | markdownify
+            | replace: 'img src="/', 'img src="$SITE_GOES_HERE'
+            | replace: '$SITE_GOES_HERE', url_prefix
```

Comment:
BTW: Liquid (Jekyll's template language) is simple and a bit wonky, so you have to dance around with stuff like this. We want to match on the `img src="` and also keep it. I don't think it can get any more elegant, as it's just too simple.

Reply:
Ugh, what a hack -- string replacement in free-form fulltext HTML? Also, site-relative `<img href>` URLs are basic HTML, and in fact preferable to make downloading/mirroring actually work. I.e. if a feed reader actually downloads everything (which many Android apps do to make reading available offline), this *breaks* images for them.

Reply:
Feel free to find a better solution. There isn't one. (This is Liquid, which is extremely limited.)

https://learn.cloudcannon.com/jekyll-cheat-sheet/
https://github.com/Shopify/liquid/wiki/Liquid-for-Designers

Reply:
Right, I don't doubt that *if* you need to replace img src tags, then that is the simplest possible mechanics. Resolving this comment, as we have another one for that "if" question.